build ios version fails

Following the documented procedure for building an ios library, I fail with the following issue.
I am not sure if we have this issue with the previous version. I have tested this with the latest version only.


CompileC ios-build/nng.build/Debug-iphoneos/nng.build/Objects-normal/arm64/xsub.o src/protocol/pubsub0/xsub.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/shmuel/dev/nng
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -target arm64-apple-ios9.0 -fmessage-length=135 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fcolor-diagnostics -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -DCMAKE_INTDIR=\"Debug-iphoneos\" -DNNG_ENABLE_STATS -DNNG_HAVE_STRLCPY=1 -DNNG_HAVE_STRNLEN=1 -DNNG_HAVE_STRCASECMP=1 -DNNG_HAVE_STRNCASECMP=1 -D_GNU_SOURCE -D_REENTRANT -D_THREAD_SAFE -D_POSIX_PTHREAD_SEMANTICS -DNNG_HAVE_LOCKF=1 -DNNG_HAVE_FLOCK=1 -DNNG_HAVE_GETRANDOM=1 -DNNG_HAVE_ARC4RANDOM=1 -DNNG_HAVE_CLOCK_GETTIME=1 -DNNG_HAVE_SEMAPHORE_PTHREAD=1 -DNNG_HAVE_PTHREAD_ATFORK_PTHREAD=1 -DNNG_HAVE_PTHREAD_SET_NAME_NP=1 -DNNG_HAVE_PTHREAD_SETNAME_NP=1 -DNNG_HAVE_LIBNSL=1 -DNNG_HAVE_LIBSOCKET=1 -DNNG_HAVE_LIBATOMIC=1 -DNNG_HAVE_UNIX_SOCKETS=1 -DNNG_HAVE_BACKTRACE=1 -DNNG_HAVE_MSG_CONTROL=1 -DNNG_HAVE_KQUEUE=1 -DNNG_HAVE_GETPEEREID=1 -DNNG_HAVE_LOCALPEERCRED=1 -DNNG_HAVE_LOCALPEERPID=1 -DNNG_HAVE_STDATOMIC=1 -DNNG_HAVE_BUS0 -DNNG_HAVE_PAIR0 -DNNG_HAVE_PAIR1 -DNNG_HAVE_PUSH0 -DNNG_HAVE_PULL0 -DNNG_HAVE_PUB0 -DNNG_HAVE_SUB0 -DNNG_HAVE_REQ0 -DNNG_HAVE_REP0 -DNNG_HAVE_SURVEYOR0 -DNNG_HAVE_RESPONDENT0 -DNNG_TRANSPORT_INPROC -DNNG_TRANSPORT_IPC -DNNG_TRANSPORT_TCP -DNNG_TRANSPORT_TLS -DNNG_TRANSPORT_WS -DNNG_SUPP_HTTP -DNNG_STATIC_LIB -DNNG_PRIVATE -DNNG_MAX_TASKQ_THREADS=16 -DNNG_HIDDEN_VISIBILITY -DNNG_PLATFORM_POSIX -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.2.sdk -fstrict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -fembed-bitcode -I/Users/shmuel/dev/nng/ios-build/Debug/include -I/Users/shmuel/dev/nng/src -I/Users/shmuel/dev/nng/include -I/Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/DerivedSources-normal/arm64 -I/Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/DerivedSources/arm64 -I/Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/DerivedSources -F/Users/shmuel/dev/nng/ios-build/Debug -Wall -Wextra -fno-omit-frame-pointer -std=gnu99 -MMD -MT dependencies -MF /Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/Objects-normal/arm64/xsub.d --serialize-diagnostics /Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/Objects-normal/arm64/xsub.dia -c /Users/shmuel/dev/nng/src/protocol/pubsub0/xsub.c -o /Users/shmuel/dev/nng/ios-build/nng.build/Debug-iphoneos/nng.build/Objects-normal/arm64/xsub.o

** BUILD FAILED **


The following build commands failed:
	CompileC ios-build/nng.build/Debug-iphoneos/nng.build/Objects-normal/arm64/posix_thread.o src/platform/posix/posix_thread.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

Found this
```bash
/Users/mikhaildemidov/Downloads/nng0/nng/src/platform/posix/posix_thread.c:261:3: error: implicit declaration of function 'pthread_set_name_np' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                pthread_set_name_np(pthread_self(), name);
                ^
/Users/mikhaildemidov/Downloads/nng0/nng/src/platform/posix/posix_thread.c:261:3: note: did you mean 'pthread_setname_np'?
In file included from /Users/mikhaildemidov/Downloads/nng0/nng/src/platform/posix/posix_thread.c:13:
In file included from /Users/mikhaildemidov/Downloads/nng0/nng/src/core/nng_impl.h:27:
In file included from /Users/mikhaildemidov/Downloads/nng0/nng/src/core/platform.h:537:
In file included from /Users/mikhaildemidov/Downloads/nng0/nng/src/platform/posix/posix_impl.h:52:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.2.sdk/usr/include/pthread.h:510:5: note: 'pthread_setname_np' declared here
int     pthread_setname_np(const char*);
        ^
/Users/mikhaildemidov/Downloads/nng0/nng/src/platform/posix/posix_thread.c:263:10: error: implicit declaration of function 'pthread_set_name_np' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                pthread_set_name_np(thr->tid, name);
```

--- a/src/platform/posix/posix_thread.c
+++ b/src/platform/posix/posix_thread.c
@@ -256,13 +256,7 @@ nni_plat_thr_is_self(nni_plat_thr *thr)
 void
 nni_plat_thr_set_name(nni_plat_thr *thr, const char *name)
 {
-#if defined(NNG_HAVE_PTHREAD_SET_NAME_NP)
-	if (thr == NULL) {
-		pthread_set_name_np(pthread_self(), name);
-	} else {
-        	pthread_set_name_np(thr->tid, name);
-	}
-#elif defined(NNG_HAVE_PTHREAD_SETNAME_NP)
+#if defined(NNG_HAVE_PTHREAD_SETNAME_NP)
 #if defined(__APPLE__)
 	// Darwin is weird, it can only set the name of pthread_self.
 	if ((thr == NULL) || (pthread_self() == thr->tid)) {
@@ -270,11 +264,17 @@ nni_plat_thr_set_name(nni_plat_thr *thr, const char *name)
 	}
 #else
 	if (thr == NULL) {
-                pthread_setname_np(pthread_self(), name);
+		pthread_setname_np(pthread_self(), name);
 	} else {
-                pthread_setname_np(thr->tid, name);
+		pthread_setname_np(thr->tid, name);
 	}
 #endif
+#elif defined(NNG_HAVE_PTHREAD_SET_NAME_NP)
+	if (thr == NULL) {
+		pthread_set_name_np(pthread_self(), name);
+	} else {
+        	pthread_set_name_np(thr->tid, name);
+	}
 #endif
 }
 

